src/Api/Billing/Controllers/InvoicesController.cs

```diff
@@ -0,0 +1,42 @@
+﻿using Bit.Core.AdminConsole.Entities;
+using Bit.Core.Billing.Models.Api.Requests.Organizations;
+using Bit.Core.Context;
+using Bit.Core.Repositories;
+using Bit.Core.Services;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Bit.Api.Billing.Controllers;
+
+[Route("invoices")]
+[Authorize("Application")]
+public class InvoicesController : BaseBillingController
+{
+    [HttpPost("preview-organization")]
+    public async Task<IResult> PreviewInvoiceAsync(
+        [FromBody] PreviewOrganizationInvoiceRequestBody model,
+        [FromServices] ICurrentContext currentContext,
+        [FromServices] IOrganizationRepository organizationRepository,
+        [FromServices] IPaymentService paymentService)
+    {
+        Organization organization = null;
```

Comment:
Non-blocker: consider using the early return pattern here to avoid nested conditionals and the null conditional operator.

```suggestion
        if (model.OrganizationId != default)
        {
            var invoice = await paymentService.PreviewInvoiceAsync(model, null, null);

            return TypedResults.Ok(invoice);
        }

        if (!await currentContext.EditPaymentMethods(model.OrganizationId))
        {
            return Error.Unauthorized();
        }

        var organization = await organizationRepository.GetByIdAsync(model.OrganizationId);

        if (organization == null)
        {
            return Error.NotFound();
        }

        var organizationInvoice = await paymentService.PreviewInvoiceAsync(model, organization.GatewayCustomerId,
            organization.GatewaySubscriptionId);

        return TypedResults.Ok(organizationInvoice);

```